src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSource.cs

```diff
@@ -9,24 +9,24 @@ namespace System.Diagnostics
 {
     public partial class DiagnosticListener : System.Diagnostics.DiagnosticSource, System.IDisposable, System.IObservable<System.Collections.Generic.KeyValuePair<string, object>>
     {
-        public DiagnosticListener(string name) { }
+        public DiagnosticListener(string? name) { }
```

Comment:
@noahfalk, does it make sense for DiagnosticListener to accept a null name? The code allows it today, which also means the Name property can return null (and so can ToString()).

Can you also take a look at the rest of these annotations to see whether they make sense (both those that have been added and places they haven't)?

Reply:
@stephentoub - Sorry, this slipped by! It doesn't make sense to me to have a null name and I am not aware of any scenario that does this. Of course I also can't rule out that someone did it anyways.

The intent is that tools/libraries that consume telemetry would iterate the list of listeners and subscribe to the ones they were interested in. The name is their sole mechanism to determine what is/isn't interesting. Technically an instrumented library and tool author could both agree that 'null' is the sentinel name they would use, but I would never recommend it.

Reply:
Going to update to non null